client/k8s/k8s.go

```diff
@@ -0,0 +1,81 @@
+package k8s
+
+import (
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	// Maximum QPS to the master from this client.
+	MaxQPS = 100
+	// Maximum burst for throttle.
+	MaxBurst = 100
+)
+
+type TLSClientConfig struct {
+	// Files containing keys/certificates.
+	CertFile string
+	KeyFile  string
+	CAFile   string
+}
+
+type Config struct {
+	InCluster bool
+	Address   string
+	TLSClientConfig
+	inClusterConfigProvider func() (*rest.Config, error)
+}
+
+func getInClusterConfig(config Config) (*rest.Config, error) {
+	if config.inClusterConfigProvider == nil {
+		config.inClusterConfigProvider = rest.InClusterConfig
+	}
+
+	return config.inClusterConfigProvider()
+}
+
+func newRawClientConfig(config Config) *rest.Config {
+	tlsClientConfig := rest.TLSClientConfig{
+		CertFile: config.CertFile,
+		KeyFile:  config.KeyFile,
+		CAFile:   config.CAFile,
+	}
+	rawClientConfig := &rest.Config{
+		Host:            config.Address,
+		QPS:             MaxQPS,
+		Burst:           MaxBurst,
+		TLSClientConfig: tlsClientConfig,
+	}
+
+	return rawClientConfig
+}
+
+func getRawClientConfig(config Config) (*rest.Config, error) {
+	var rawClientConfig *rest.Config
+	var err error
+
+	if config.InCluster {
+		rawClientConfig, err = getInClusterConfig(config)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		rawClientConfig = newRawClientConfig(config)
+	}
+
+	return rawClientConfig, nil
+}
+
+func NewClient(config Config) (kubernetes.Interface, error) {
```

Comment:
When this k8s client magic is state of the art and we want to use this in different projects it would be cool to prevent the copy/paste and add some package to microkit.